### PR TITLE
Patrol planner

### PR DESCRIPTION
**Pull request by jrbaxter0**
A simple, configurable planner that has a chance at working. Tuning will be required on the hardware, but it works well enough in sim

### CODE REVIEW

File: rktl_planner/nodes/patrol_planner [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+        NORTH_HEADING_ERROR = sad(yaw, 0.0)
+        SOUTH_HEADING_ERROR = sad(yaw, math.pi)
+        EAST_HEADING_ERROR = sad(yaw, -math.pi/2.0)
+        WEST_HEADING_ERROR = sad(yaw, +math.pi/2.0)
+
+        # Prep command message for publish
+        cmd_msg = ControlCommand()
+
+        # check to see if an existing attempt has timed out
+        if self.attempt_start_time is not None and (rospy.Time.now() - self.attempt_start_time).to_sec() > self.ATTEMPT_TIMEOUT:
```
Comment (hmccarty) [inline rktl_planner/nodes/patrol_planner:104]: Picky, but maybe break into 2 lines?

File: rktl_planner/nodes/patrol_planner [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+            return
+
+        # extract ball position and velocity, adding look-ahead time
+        bx, by = self.ball_position
+        vbx, vby = self.ball_velocity
+        bx += vbx * self.LOOKAHEAD_TIME
+        by += vby * self.LOOKAHEAD_TIME
+
+        # check if we should be offensive or defensive
+        if bx > 0.0:    # offensive
```
Comment (hmccarty) [inline rktl_planner/nodes/patrol_planner:123]: Maybe we look into adding a slight offset. Like we may not want to be offensive until the ball is in the last 1/4th of the field.

Comment (hmccarty) [reply rktl_planner/nodes/patrol_planner:123]: At least might be something to parametrize

**Review by hmccarty — CHANGES_REQUESTED** [review]
Most of the comments are style-focused since we've already seen it work fairly well irl.

File: rktl_planner/nodes/patrol_planner (hunk 2 above)
Comment (jrbaxter0) [reply rktl_planner/nodes/patrol_planner]: done

**Review by hmccarty — APPROVED** [review]

**Pull request merged**